### PR TITLE
test(router): add tests for router.d.ts

### LINKED ISSUE #3282: router.d.ts is untested
We need some sanity test for the router .d.ts file. The `gulp test.typings` target does this for the `angular2.d.ts`.
I did not update router on DefinitelyTyped for alpha.32 because it fails the definitelyTyped tests, for example, the Instruction class is referenced but not defined.
We can't release typings for router until this is fixed.

### PR DESCRIPTION
**Pull request by btford**
~~~I need some help from @alexeagle to get this in.~~~

~~~I'm still getting:~~~

```
[gulp] Starting 'test.typings'...
/Users/bford/Development/angular/dist/docs/typings/angular2/angular2.d.ts(6563,3): 2300 Duplicate identifier 'export='.
[gulp] 'test.typings' errored after 1.34 s /Users/bford/Development/angular/dist/docs/typings/angular2/angular2.d.ts(6563,3): 2300 Duplicate identifier 'export='.
/Users/bford/Development/angular/dist/docs/typings/angular2/router.d.ts(468,3): 2300 Duplicate identifier 'export='.
typing_spec/router_spec.ts(5,64): 2307 Cannot find module 'angular2/router'.
```

Should be good to go!

Closes #3282

### CODE REVIEW

File: modules/angular2/src/router/lifecycle_annotations.ts [hunk 1]
```diff
@@ -14,4 +14,4 @@ export {
   onDeactivate
 } from './lifecycle_annotations_impl';
 
-export var CanActivate = makeDecorator(CanActivateAnnotation);
+export var CanActivate: Function = makeDecorator(CanActivateAnnotation);
```
Comment (alexeagle) [inline modules/angular2/src/router/lifecycle_annotations.ts]: can we type this more? how does this look to an IDE user?
See ClassDecorator from lib.d.ts

File: typing_spec/router_spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+@Component({
+  selector: 'my-app'
+})
+@View({
+	template: '<h1>Hello {{ name }}</h1><router-outlet></router-outlet>',
+  directives: routerDirectives
+})
+@RouteConfig([
```
Comment (alexeagle) [inline typing_spec/router_spec.ts]: in a later change: we should add a spec that uses the non-decorator DSL and confirm it's correctly typed as well

File: modules/angular2/src/router/lifecycle_annotations.ts (hunk 1 above)
Comment (btford) [inline modules/angular2/src/router/lifecycle_annotations.ts]: :+1:

**Comment by btford** [comment]
Pushed to presubmit.

**Comment by btford** [comment]
someday, presubmit will pass :cat2:

**Comment by btford** [comment]
Landed as 450d3630cc2eb5afdbb535334d647a830badf91c.

**Pull request closed without merging**